nomos-services/data-availability/sampling/src/storage/adapters/rocksdb.rs

```diff
@@ -36,22 +40,44 @@ where
     type Settings = RocksAdapterSettings;
 
     async fn new(
-        _settings: Self::Settings,
-        _storage_relay: OutboundRelay<<StorageService<Self::Backend> as ServiceData>::Message>,
+        settings: Self::Settings,
+        storage_relay: OutboundRelay<<StorageService<Self::Backend> as ServiceData>::Message>,
     ) -> Self {
         Self {
-            _settings,
-            _storage_relay,
+            settings,
+            storage_relay,
             blob: PhantomData,
         }
     }
 
     async fn get_blob(
         &self,
-        _blob_id: <Self::Blob as Blob>::BlobId,
-        _column_idx: ColumnIndex,
+        blob_id: <Self::Blob as Blob>::BlobId,
+        column_idx: ColumnIndex,
     ) -> Result<Option<Self::Blob>, DynError> {
-        todo!()
+        let (reply_tx, reply_rx) = tokio::sync::oneshot::channel();
+        self.storage_relay
+            .send(StorageMsg::Load {
+                key: key_bytes(DA_VERIFIED_KEY_PREFIX, &blob_id),
+                reply_channel: reply_tx,
+            })
+            .await
+            .expect("failed to send Load message to storage relay");
+
+        if reply_rx.await?.is_some() {
+            let blob_bytes = load_blob(
+                self.settings.blob_storage_directory.clone(),
+                blob_id.as_ref(),
+                &column_idx.to_be_bytes(),
+            )
+            .await
+            .unwrap();
+            Ok(S::deserialize(blob_bytes)
+                .map(|blob| Some(blob))
+                .unwrap_or_default())
```

Comment:
Just a suggestion for grouping deserialization with the loading. The `load_blob().unwrap()` can be changed to `?`.
```suggestion
            let blob = S::deserialize(
                load_blob(
                    self.settings.blob_storage_directory.clone(),
                    blob_id.as_ref(),
                    &column_idx.to_be_bytes(),
                )
                .await?,
            )
            .map(|b| Some(b))
            .unwrap_or_default();
            Ok(blob)
```

Reply:
Actually your variant is better, only lets change the unwrap and it's good to go!